azure.yaml

```diff
@@ -1,34 +1,80 @@
-# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
+trigger:
+- main
 
-name: azure-search-openai-demo
-metadata:
-  template: azure-search-openai-demo@0.0.2-beta
-services:
-  backend:
-    project: ./app/backend
-    language: java
-    host: appservice
-    hooks:
-      prepackage:
-        windows:
-          shell: pwsh
-          run:  cd ../frontend;npm install;npm run build
-          interactive: true
-          continueOnError: false
-        posix:
-          shell: sh
-          run:  cd ../frontend;npm install;npm run build
-          interactive: true
-          continueOnError: false
-hooks:
-    postprovision:
-      windows:
-        shell: pwsh
-        run: ./scripts/prepdocs.ps1
-        interactive: true
-        continueOnError: false
-      posix:
-        shell: sh
-        run: ./scripts/prepdocs.sh
-        interactive: true
-        continueOnError: false
+pool:
+  vmImage: 'ubuntu-latest'
+
+jobs:
+- job: Build
+  steps:
+  - task: Npm@1
+    displayName: 'Npm Install'
+    inputs:
+      command: 'install'
+      workingDir: 'app/frontend/'
+  - task: Npm@1
+    displayName: 'Npm run build'
+    inputs:
+      command: 'custom'
+      workingDir: 'app/frontend/'
+      customCommand: 'run build'
+    
+    # build springboot app with Maven
+  - task: Maven@4
+    displayName: 'Build Springboot App' 
+    inputs:
+      mavenPomFile: 'app/backend/pom.xml'
+      goals: 'package'
+      javaHomeOption: 'JDKVersion'
+      jdkVersionOption: '1.17'
+      jdkArchitectureOption: 'x64'
+      publishJUnitResults: false
+      mavenVersionOption: 'Default'
+      mavenAuthenticateFeed: false
+      effectivePomSkip: false
+      sonarQubeRunAnalysis: false
+
+    # get jar name from pom.xml and set it as output variable so Deploy job and use it
+  - powershell: |

```

Comment:
any chance to convert this to shell? typically java developer are more inclined towards linux shell rather than power shell. an example could be:
artifactid=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)
 jarversion=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
 originaljarname="$artifactid-$jarversion.jar" 
echo "##vso[task.setvariable variable=jarfilename]$(echo $originaljarname)"